translator/reader/reader.cpp

```diff
@@ -966,42 +947,30 @@ void UsdArnoldReaderThreadContext::SetReader(UsdArnoldReader *r)
 }
 void UsdArnoldReaderThreadContext::AddNodeName(const std::string &name, AtNode *node)
 {
-    if (_addNodeNameLock)
-        AiCritSecEnter(&_addNodeNameLock);
+    if (_dispatcher)
+        _addNodeNameLock.lock();
     _nodeNames[name] = node;
-    if (_addNodeNameLock)
-        AiCritSecLeave(&_addNodeNameLock);
+    if (_dispatcher)
+        _addNodeNameLock.unlock();
 }
 
 void UsdArnoldReaderThreadContext::SetDispatcher(WorkDispatcher *dispatcher)
 {
 
     _dispatcher = dispatcher;
-    if (_dispatcher) {
-        if (!_createNodeLock) 
-            AiCritSecInit((void **)&_createNodeLock);
-        if (!_addConnectionLock)
-            AiCritSecInit((void **)&_addConnectionLock);
-        if (!_addNodeNameLock)
-            AiCritSecInit((void **)&_addNodeNameLock);
-    }
 }
 
 AtNode *UsdArnoldReaderThreadContext::CreateArnoldNode(const char *type, const char *name)
 {    
-    AtNode *node = AiNode(_reader->GetUniverse(), type, name, _reader->GetProceduralParent());
+    AtNode *node = AiNode(_reader->GetUniverse(), AtString(type), AtString(name), _reader->GetProceduralParent());
     // All shape nodes should have an id parameter if we're coming from a parent procedural
     if (_reader->GetProceduralParent() && AiNodeEntryGetType(AiNodeGetNodeEntry(node)) == AI_NODE_SHAPE) {
         AiNodeSetUInt(node, str::id, _reader->GetId());
     }
-    
-    if (_createNodeLock)
-        AiCritSecEnter(&_createNodeLock);
+
+    std::lock_guard<AtMutex> guard(_createNodeLock);
```

Comment:
We're just missing a if(dispatcher) to lock `_createNodeLock` , as for the 2 other locks. This way we won't be locking in the other modes.